src/Microsoft.Diagnostics.Monitoring.RestServer/MetricsService.cs

```diff
@@ -44,11 +46,23 @@ protected override Task ExecuteAsync(CancellationToken stoppingToken)
                         //TODO In multi-process scenarios, how do we decide which process to choose?
                         //One possibility is to enable metrics after a request to begin polling for metrics
                         IProcessInfo pi = await _services.GetProcessAsync(filter: null, stoppingToken);
-                        await _pipeProcessor.Process(pi.Client, pi.Pid, Timeout.InfiniteTimeSpan, stoppingToken);
+                        _counterPipeline = new EventCounterPipeline(pi.Client, new EventPipeCounterPipelineSettings
+                        {
+                            CounterGroups = Array.Empty<EventPipeCounterGroup>(),
+                            Duration = Timeout.InfiniteTimeSpan,
+                            ProcessId = pi.Pid,
+                            RefreshInterval = TimeSpan.FromSeconds(_metricsConfiguration.UpdateIntervalSeconds)
+                        }, metricsLogger: new[] { new MetricsLogger(_store.MetricsStore) });
+
+                        await _counterPipeline.RunAsync(stoppingToken);
                     }
-                    catch(Exception e) when (!(e is OperationCanceledException))
+                    catch(Exception)
```

Comment:
You should allow OperationCanceledException to unwind out of here so that the ExecuteAsync method can be cancelled.